### PR TITLE
add SYMBOL_VISIBILITY cache variable to match scons interface.

### PR DESCRIPTION
**Pull request by enetheru**
Bring the 'cmake -LH' output inline with the scons output.
I dont know what the policy on adding notes in comments are.
I can ammend the commit to remove them if they are unwanted.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -49,6 +49,18 @@ option(GENERATE_TEMPLATE_GET_NODE "Generate a template version of the Node class
...
 
+set( SYMBOL_VISIBILITY "hidden" CACHE STRING "Symbols visibility on GNU platforms. Use 'auto' to apply the default value. (auto|visible|hidden)")
+set_property( CACHE SYMBOL_VISIBILITY PROPERTY STRINGS "auto;visible;hidden" )
+
+# CXX_VISIBILITY_PRESET supported values are: default, hidden, protected, and internal
+# which is inline with the gcc -fvisibility=
+# https://gcc.gnu.org/onlinedocs/gcc/Code-Gen-Options.html
+# To match the scons options we need to change the text to match the -fvisibility flag
+# it is probably worth another PR which changes both to use the flag options
+if( ${SYMBOL_VISIBILITY} STREQUAL "auto" OR ${SYMBOL_VISIBILITY} STREQUAL "visible" )
```
Comment (dsnopek) [inline CMakeLists.txt:60]: I think we need to decide if we are going to standardize on having a `GODOT_` prefix for all our options (and so `GODOT_SYMBOL_VISIBILITY`), or not.

What do most other cmake projects do?

**Review by dsnopek — COMMENTED** [review]
Thanks!

> I dont know what the policy on adding notes in comments are.
> I can ammend the commit to remove them if they are unwanted.

The comment is good.

File: CMakeLists.txt (hunk 1 above)
Comment (Bromeon) [reply CMakeLists.txt:60]: [SFML](https://github.com/SFML/SFML/blob/master/CMakeLists.txt) also prefixes options with `SFML_` (unless they're standardized, e.g. `CMAKE_BUILD_TYPE`).
Some macros also have it: `sfml_set_option`.

Comment (enetheru) [reply CMakeLists.txt:60]: yes prefixing variables is the norm in cmake projects, would you like me to go back and change it, and then submit a PR bringing other options into line?

Comment (dsnopek) [reply CMakeLists.txt:60]: That would be great!

Comment (enetheru) [reply CMakeLists.txt:60]: Execllent, to be clear we will be prefixing everything with GODOT_ yeah ? not GODOT_CPP_?

Comment (dsnopek) [reply CMakeLists.txt]: Yeah, let's do `GODOT_` - we're already mostly using that one (with a couple exceptions), and there really isn't a world where you have Godot and godot-cpp in the same codebase, so I don't think we'll conflict

**Review by dsnopek — APPROVED** [review]
Looks great, thanks!

**Comment by dsnopek** [comment]
Ah, sorry, I didn't notice this when I approved: you need to squash this down to a single commit before I can merge it.

This is part of Godot's development workflow: https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#modifying-a-pull-request

**Comment by enetheru** [comment]
Apologies, commits have been squashed and I'll remember it for next time.

**Comment by dsnopek** [comment]
Thanks!

**Comment by dsnopek** [comment]
Cherry-picked for 4.2 in PR https://github.com/godotengine/godot-cpp/pull/1631

**Comment by dsnopek** [comment]
Cherry-picked for 4.3 in PR https://github.com/godotengine/godot-cpp/pull/1632

**Pull request merged**